frontends/precinct-scanner/src/app.test.tsx

```diff
@@ -120,6 +120,43 @@ test('shows setup card reader screen when there is no card reader', async () =>
   await screen.findByText('Card Reader Not Detected');
 });
 
+test('initializes app with stored state', async () => {
+  const logger = fakeLogger();
+  const card = new MemoryCard();
+  const hardware = MemoryHardware.buildStandard();
+  const storage = new MemoryStorage();
```

Comment:
I played around with this a bit to see why the test might not be working. I realized that I could reproduce the behavior in kiosk-browser but not Chrome, which gave me a clue to look at the difference between `KioskStorage` (which is truly async) and `LocalStorage` (which is fake async). Making `LocalStorage` truly async and fiddling with timeouts can cause the same race condition to happen as we're seeing with kiosk-browser. I think tweaking MemoryStorage in a similar way will make your test fail, though it's a very specific tweak so I'm not sure if it's something we want to do permanently.

```diff
diff --git a/libs/utils/src/Storage/local_storage.ts b/libs/utils/src/Storage/local_storage.ts
index f2b0ba30d..b53e03619 100644
--- a/libs/utils/src/Storage/local_storage.ts
+++ b/libs/utils/src/Storage/local_storage.ts
@@ -12,8 +12,12 @@ export class LocalStorage implements Storage {
    */
   async get(key: string): Promise<unknown> {
     assert(typeof key === 'string');
-    const value = window.localStorage.getItem(key);
-    return value ? JSON.parse(value) : undefined;
+    return await new Promise((resolve) => {
+      setTimeout(() => {
+        const value = window.localStorage.getItem(key
);
+        resolve(value ? JSON.parse(value) : undefined
);
+      }, 10);
+    });
   }
 
   /**
@@ -21,7 +25,12 @@ export class LocalStorage implement
s Storage {
    */
   async set(key: string, value: unknown): Promise<voi
d> {
     assert(typeof key === 'string');
-    window.localStorage.setItem(key, JSON.stringify(v
alue));
+    return await new Promise((resolve) =>
+      setTimeout(() => {
+        window.localStorage.setItem(key, JSON.stringi
fy(value));
+        resolve();
+      }, 0)
+    );
   }
 
   /**
```